src/main/java/com/jcraft/jsch/KeyPair.java

```diff
@@ -175,6 +175,15 @@ public void writePrivateKey(OutputStream out, byte[] passphrase){
 
   abstract byte[] getKeyTypeName();
   public abstract int getKeyType();
+  
+  public String getKeyTypeName(String encoding) {
```

Comment:
This part seems unrelated to the logging changes?

Reply:
It is. Shouldn't be part of the commit, either.